src/tools/compiletest/src/header.rs

```diff
@@ -636,8 +636,12 @@ impl TestProps {
 
     pub fn pass_mode(&self, config: &Config) -> Option<PassMode> {
         if !self.ignore_pass {
-            if let (mode @ Some(_), Some(_)) = (config.force_pass_mode, self.pass_mode) {
-                return mode;
+            if let (mode @ Some(_), Some(pm_file)) = (config.force_pass_mode, self.pass_mode) {
+                if pm_file != PassMode::RunFail {
+                    // HACK(Centril): See #66929. Ensure that `--pass check`
```

Comment:
`PassMode::RunFail`? O_O
cc https://github.com/rust-lang/rust/pull/64158
@tmandry, run-*fail* tests are clearly not pass tests and should not be affected by `--pass` options at all, similarly to other tests failing at compile time.

Can we get a proper solution instead of hacks?
It's not super urgent and the tests can be moved into the run-fail directory in the meantime if we need to wait for more than a ~week.

Reply:
I was thinking the same thing, but I wanted to write a quick hack to make `--pass check` work again, which seemed the higher order priority. Practically speaking this implementation does achieve run-fail not being affected by the `--pass` option but I would agree that there's a cleaner way to do it code-wise. Perhaps we should file an issue for improving the impl for now?

Reply:
I'm personally ok with the issue being unfixed for a week, which should be enough for the improved version.
If someone wants it sooner, feel free to r+.

Reply:
Sorry for misunderstanding, I was actually interpreting "pass" more in the sense of "compiler pass." As in this is a "test pass" of a certain category of tests.

Reply:
Or in the specific test case (which I probably had in mind here), "this test passes only when we run this program and it fails."